x/distribution/keeper/calculation_test.go

```diff
@@ -0,0 +1,414 @@
+	require.True(t, rewards.IsZero())
+
+	// start out block height
+	ctx = ctx.WithBlockHeight(ctx.BlockHeight() + 3)
+
+	// slash the validator by 50%
+	sk.Slash(ctx, valConsAddr1, ctx.BlockHeight(), 100, sdk.NewDecWithPrec(5, 1))
+
+	// fetch the validator again
+	val = sk.Validator(ctx, valOpAddr1)
+
+	// increase block height
+	ctx = ctx.WithBlockHeight(ctx.BlockHeight() + 3)
+
+	// allocate some rewards
+	tokens := sdk.DecCoins{{staking.DefaultBondDenom, sdk.NewDec(10)}}
+	k.AllocateTokensToValidator(ctx, val, tokens)
+
+	// slash the validator by 50% again
+	sk.Slash(ctx, valConsAddr1, ctx.BlockHeight(), 50, sdk.NewDecWithPrec(5, 1))
+
+	// fetch the validator again
+	val = sk.Validator(ctx, valOpAddr1)
+
+	// increase block height
+	ctx = ctx.WithBlockHeight(ctx.BlockHeight() + 3)
+
+	// allocate some more rewards
+	k.AllocateTokensToValidator(ctx, val, tokens)
+
+	// end period
+	endingPeriod = k.incrementValidatorPeriod(ctx, val)
+
+	// calculate delegation rewards
+	rewards = k.calculateDelegationRewards(ctx, val, del, endingPeriod)
+
+	// rewards should be half the tokens
+	require.Equal(t, sdk.DecCoins{{staking.DefaultBondDenom, sdk.NewDec(10)}}, rewards)
+
+	// commission should be the other half
+	require.Equal(t, sdk.DecCoins{{staking.DefaultBondDenom, sdk.NewDec(10)}}, k.GetValidatorAccumulatedCommission(ctx, valOpAddr1))
+}
+
+func TestCalculateRewardsMultiDelegator(t *testing.T) {
+	ctx, _, k, sk, _ := CreateTestInputDefault(t, false, 1000)
+	sh := staking.NewHandler(sk)
+
+	// initialize state
+	k.SetOutstandingRewards(ctx, sdk.DecCoins{})
+
+	// create validator with 50% commission
+	commission := staking.NewCommissionMsg(sdk.NewDecWithPrec(5, 1), sdk.NewDecWithPrec(5, 1), sdk.NewDec(0))
+	msg := staking.NewMsgCreateValidator(valOpAddr1, valConsPk1,
+		sdk.NewCoin(staking.DefaultBondDenom, sdk.NewInt(100)), staking.Description{}, commission)
+	require.True(t, sh(ctx, msg).IsOK())
+
+	// end block to bond validator
+	staking.EndBlocker(ctx, sk)
+
+	// fetch validator and delegation
+	val := sk.Validator(ctx, valOpAddr1)
+	del1 := sk.Delegation(ctx, sdk.AccAddress(valOpAddr1), valOpAddr1)
+
+	// allocate some rewards
+	tokens := sdk.DecCoins{{staking.DefaultBondDenom, sdk.NewDec(10)}}
+	k.AllocateTokensToValidator(ctx, val, tokens)
+
+	// second delegation
+	msg2 := staking.NewMsgDelegate(sdk.AccAddress(valOpAddr2), valOpAddr1, sdk.NewCoin(staking.DefaultBondDenom, sdk.NewInt(100)))
+	require.True(t, sh(ctx, msg2).IsOK())
+	del2 := sk.Delegation(ctx, sdk.AccAddress(valOpAddr2), valOpAddr1)
+
+	// fetch updated validator
+	val = sk.Validator(ctx, valOpAddr1)
+
+	// end block
+	staking.EndBlocker(ctx, sk)
+
+	// allocate some more rewards
+	k.AllocateTokensToValidator(ctx, val, tokens)
+
+	// end period
+	endingPeriod := k.incrementValidatorPeriod(ctx, val)
+
+	// calculate delegation rewards for del1
+	rewards := k.calculateDelegationRewards(ctx, val, del1, endingPeriod)
+
+	// rewards for del1 should be 7.5
```

Comment:
why 7.5? would be good to have a comment explaining the math... or better calculating the math based on earlier defined variables in this test.